message/messaging_service.cc

```diff
@@ -149,16 +149,22 @@ const size_t PER_SHARD_CONNECTION_COUNT = 2;
 const size_t PER_TENANT_CONNECTION_COUNT = 3;
 
 bool operator==(const netw::msg_addr& x, const netw::msg_addr& y) noexcept {
-    // Ignore cpu id for now since we do not really support shard to shard connections
-    return x.addr == y.addr;
+     // Ignore cpu id for now since we do not really support shard to shard connections
+    return x.addr == y.addr && (!x.host_id || !y.host_id || x.host_id == y.host_id);
 }
 
-bool operator<(const netw::msg_addr& x, const netw::msg_addr& y) noexcept {
+std::strong_ordering msg_addr::operator<=>(const msg_addr& o) const noexcept {
```

Comment:
Is `operator==` (above) still necessary if there is a spaceship operator available?

Reply:
yes, see https://en.cppreference.com/w/cpp/language/default_comparisons
```
The three-way comparison function (whether defaulted or not) is called whenever values are compared using <, >, <=, >=, or <=> and overload resolution selects this overload.

The equality comparison function (whether defaulted or not) is called whenever values are compared using == or != and overload resolution selects this overload.
```

Reply:
ok